config/float32_test.go

```diff
@@ -1,201 +1,153 @@
 )
 
-var _ = Describe("func GetFloat32()", func() {
-	It("returns a positive float value", func() {
-		b := Map{"<key>": String("123.45")}
-
-		v, ok, err := GetFloat32(b, "<key>")
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(v).To(BeNumerically("~", float32(123.45)))
-		Expect(ok).To(BeTrue())
-	})
-
-	It("returns a negative float value", func() {
-		b := Map{"<key>": String("-123.45")}
+var _ = Describe("func AsFloat32()", func() {
+	It("returns an float32 value", func() {
+		b := Map{"<key>": String("-50.5")}
 
-		v, ok, err := GetFloat32(b, "<key>")
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(v).To(BeNumerically("~", float32(-123.45)))
-		Expect(ok).To(BeTrue())
+		v := AsFloat32(b, "<key>")
+		Expect(v).To(BeNumerically("==", -50.5))
 	})
 
-	It("sets ok to false if the key is not defined", func() {
+	It("panics if the key is not defined", func() {
 		b := Map{}
 
-		_, ok, err := GetFloat32(b, "<key>")
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(ok).To(BeFalse())
+		Expect(func() {
+			AsFloat32(b, "<key>")
+		}).To(PanicWith(`<key> is not defined`))
 	})
 
-	It("returns an error if the value cannot be parsed", func() {
+	It("panics if the value cannot be parsed", func() {
 		b := Map{"<key>": String("<invalid>")}
 
-		_, _, err := GetFloat32(b, "<key>")
-		Expect(err).To(MatchError(`<key> is not a valid 32-bit float: strconv.ParseFloat: parsing "<invalid>": invalid syntax`))
+		Expect(func() {
+			AsFloat32(b, "<key>")
+		}).To(PanicWith(`expected <key> to be a 32-bit floating-point number: strconv.ParseFloat: parsing "<invalid>": invalid syntax`))
 	})
 })
 
-func ExampleGetFloat32() {
-	os.Setenv("FOO", "1.5")
-
-	v, ok, err := config.GetFloat32(config.Environment(), "FOO")
-	if err != nil {
-		panic(err)
-	}
-
-	if !ok {
-		fmt.Println("key is not defined!")
-	} else {
-		fmt.Printf("the value is %f!\n", v)
-	}
-
-	// Output: the value is 1.500000!
-}
-
-var _ = Describe("func GetFloat32Default()", func() {
-	It("returns a positive float value", func() {
-		b := Map{"<key>": String("123.45")}
+var _ = Describe("func AsFloat32Default()", func() {
+	It("returns an float32 value", func() {
+		b := Map{"<key>": String("-50")}
 
-		v, err := GetFloat32Default(b, "<key>", 456.78)
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(v).To(BeNumerically("~", float32(123.45)))
-	})
-
-	It("returns a negative float value", func() {
-		b := Map{"<key>": String("-123.45")}
-
-		v, err := GetFloat32Default(b, "<key>", 456.78)
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(v).To(BeNumerically("~", float32(-123.45)))
+		v := AsFloat32Default(b, "<key>", 50)
+		Expect(v).To(BeNumerically("==", -50))
 	})
 
 	It("returns the default value if the key is not defined", func() {
 		b := Map{}
 
-		v, err := GetFloat32Default(b, "<key>", 456.78)
-		Expect(err).ShouldNot(HaveOccurred())
-		Expect(v).To(BeNumerically("~", float32(456.78)))
+		v := AsFloat32Default(b, "<key>", 50)
+		Expect(v).To(BeNumerically("==", 50))
```

Comment:
Should these float tests use float looking values, `50.0`?

Reply:
Yep, will fix.